app/helpers/nav_helper.rb

```diff
@@ -6,35 +6,68 @@ def nav_items
     sample_period = Work::Period.new(community: current_community)
     sample_job = Work::Job.new(period: sample_period)
     sample_shift = Work::Shift.new(job: sample_job)
-    items = [
-      {
+    # Build up the main menu list
+    items = []
+    # check if there are any main menu items defined in Settings.
+    c_settings_topmenu = current_community.settings.top_menu_customizations
+    settings_topmenu = if c_settings_topmenu.nil?
+                         []
```

Comment:
You can simplify this part a bit. Try 

```
settings_topmenu = current_community.settings.top_menu_customizations || []
settings_topmenu = settings_topmenu.scan(%r{\[([\w\- ]+)\]\(([\w/\.:%&\?]+)\)})
```

Reply:
Thinking about this a bit more, I think this parsing logic should be in a very simple class so that we can write unit tests for it, which it definitely should have. This kind of text parsing is quite error prone so I'd want to test a few different combinations of input.

Can you give that a shot? Perhaps MenuCustomizationParser. See app/models/work/quota_calculator.rb for an example of a service class like that. The constructor argument should be the value of the setting.